### PR TITLE
Documentation on how to change matplotlib defaults

### PR DESCRIPTION
**Pull request by keeeto**
**Description of work.**
This PR adds some documentation on setting default values in Matplotlib. This is as a result of feedback from the user demo at ISIS. 
**Report to:** Nina-Juliane Steinke

**To test:**
* Build the docs-html target.
* Check the Matplotlib docs to see if the new additions make sense.

*There is no associated issue.*

*This does not require release notes* because it is a minor documentation update.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by vreimund — COMMENTED** [review]
Using 

```
mpl.rcParams['axes.color_cycle'] = cycler(color='cmyk')
```

and fixing a typo in *available* would make this work for me.

But I believe that this documentation is not needed. Matplotlib documents it already. The maintenance work will be immense -> check version -> different API ... 
I recommend closing this PR simply and referring to matplotlib's documentation!!!

https://matplotlib.org/1.5.3/api/api_changes.html

Changes in 1.5.0:

color_cycle deprecated

In light of the new property cycling feature, the Axes method set_color_cycle is now deprecated. Calling this method will replace the current property cycle with one that cycles just the given colors.

Similarly, the rc parameter axes.color_cycle is also deprecated in lieu of the new axes.prop_cycle parameter. Having both parameters in the same rc file is not recommended as the result cannot be predicted. For compatibility, setting axes.color_cycle will replace the cycler in axes.prop_cycle with a color cycle. Accessing axes.color_cycle will return just the color portion of the property cycle, if it exists.

**Comment by NickDraper** [comment]
@reimundILL While I would normally agree with you about not documenting third party code again in our systems, I just went searching for any similar documentation in the Matplotlib documentation, and I could not find anything that was close to being able to use as an equivalent reference.

**Comment by reimundILL** [comment]
My guess or I am afraid that each build machine and user machine may have a different version of matplotlib installed and then, the API changes ...

I fully agree that the documentation is user-friendly and well written. Good for education and help on publications.

**Review by DanNixon — CHANGES_REQUESTED** [review]
I agree with @reimundILL, we shouldn't be documenting our dependencies and nothing here is Mantid related.

A link to https://matplotlib.org/users/customizing.html in some kind of FAQ section is all this documentation should be.

If we think the Matplotlib documentation is lacking then we should propose changes to that.

File: docs/source/plotting/index.rst [hunk 1]
```diff
@@ -563,3 +563,66 @@ Plotting dispersion curves  on multiple panels can also be done using matplotlib
...
+    mpl.rcParams['axes.facecolor'] = (0.95, 0.95, 0.95)
+    mpl.rcParams['grid.linestyle'] = '--'
+    # Plot the data
+    fig, ax = plt.subplots()
+    ax.plot(yy)
+
+For much more on customising the graph appearance see the `Matplotlib documentation 
+<https://matplotlib.org/users/dflt_style_changes.html>`_.
+
+A list of some common properties you might want to change and the handles to set:
```
Comment (AnthonyLim23) [inline docs/source/plotting/index.rst]: is it worth saying `key words` instead of `handles`?

**Review by AnthonyLim23 — APPROVED** [review]
Most users will want simple and easy instructions on how to make changes. This documentation is a good way for users to learn how to change the way a plot looks. The majority of users would look at the matplotlib documentation and then close it as they do not understand it. They may then decide that the Mantid documentation is not worth reading (which can prevent them from using other documentation and as a result we have to answer questions that have answers in the documentation). It is therefore worth having this cheat sheet in the Mantid documentation

**Review by DanNixon — APPROVED** [review]

**Pull request merged**